services/app-api/src/handlers/audit_s3.ts

```diff
@@ -165,4 +180,67 @@ async function processDocument(
     }
 }
 
+function deduplicateDocuments(documents: AuditDocument[]): AuditDocument[] {
+    const uniqueDocuments = new Map<string, AuditDocument>()
+
+    for (const doc of documents) {
+        if (!uniqueDocuments.has(doc.s3URL)) {
+            uniqueDocuments.set(doc.s3URL, doc)
+        }
+    }
+
+    return Array.from(uniqueDocuments.values())
+}
+
+type DocumentWithAssociation = AuditDocument & {
+    associatedContract?: ContractRevisionTable
+    associatedRate?: RateRevisionTable
+}
+
+async function fetchAssociatedData(
+    store: Store,
+    documents: AuditDocument[]
+): Promise<DocumentWithAssociation[]> {
+    const results: DocumentWithAssociation[] = []
+
+    for (const doc of documents) {
+        let associatedData: ContractRevisionTable | RateRevisionTable | null =
+            null
+
+        if (doc.type === 'contractDoc' && doc.contractRevisionID) {
+            const contractResult = await store.findContractRevision(
+                doc.contractRevisionID
+            )
+            if (
+                !(contractResult instanceof Error) &&
```

Comment:
And what if they are errors? Looking at the store code I don't see us printing those errors, will this fail silently? Or is OTEL catching the errors somewhere I don't see?